src/dune_rules/ctypes_stanza.ml

```diff
@@ -0,0 +1,141 @@
+open! Stdune
+open Import
+open Dune_lang.Decoder
+
+module Build_flags_resolver = struct
+
+  module Vendored = struct
+    type t =
+      { c_flags : Ordered_set_lang.Unexpanded.t
+      ; c_library_flags : Ordered_set_lang.Unexpanded.t }
+
+    let decode =
+      fields
+        (let+ c_flags = Ordered_set_lang.Unexpanded.field "c_flags"
+          and+ c_library_flags = Ordered_set_lang.Unexpanded.field "c_library_flags" in
+          { c_flags; c_library_flags })
+  end
+
+  type t =
+    | Pkg_config
+    | Vendored of Vendored.t
+
+  let decode =
+    let vendored =
+      let+ p = Vendored.decode in
+      Vendored p
+    in
+    sum [ ("pkg_config" , return Pkg_config)
+        ; ("vendored"   , vendored ) ]
+
+  let default = Pkg_config
+end
+
+module Concurrency_policy = struct
+  type t =
+    | Sequential
+    | Unlocked
+    | Lwt_jobs
+    | Lwt_preemptive
+
+  let decode =
+    enum [ "sequential"     , Sequential
+          ; "unlocked"       , Unlocked
+          ; "lwt_jobs"       , Lwt_jobs
+          ; "lwt_preemptive" , Lwt_preemptive ]
+
+  let default = Sequential
+end
+
+module Headers = struct
+  type t =
+    | Include of string list
```

Comment:
I think this `Include` option doesn't work so well in practice. One cannot use it to include headers relative to the source, and it doesn't add dependencies automatically. Can we do without it?